api/_utils/auth/handle-verified-requests.ts

```diff
@@ -87,36 +123,44 @@ export async function handleVerifiedRequest(
         const {
           queryType,
           tree_id,
+          timestamp,
           uuid,
           username,
           amount,
         } = request.body as RequestBody;
 
         switch (queryType) {
           case "adopt":
-            if (tree_id === undefined || uuid === undefined) {
+            try {
+              validateString(tree_id, request.method, "tree_id");
+              validateString(uuid, request.method, "uuid");
+            } catch (error) {
               statusCode = 400;
-              throw new Error(
-                "POST body needs uuid (string) and tree_id (string) properties",
-              );
+              throw error;
             }
-            result = await adoptTree(tree_id, uuid);
+
+            result = await adoptTree(tree_id as string, uuid as string);
```

Comment:
Why all the casting to string?

Reply:
Because typescript is not aware that my validation function already checks for that. there is no way this code is reached if tree_id is not a string. No so nice but I don't want to redo a condition here just to please typescript.